internal/experiment/quicping/quicping.go

```diff
@@ -227,12 +227,19 @@ func (m *Measurer) Run(ctx context.Context, args *model.ExperimentArgs) error {
 	sess := args.Session
 
 	host := string(measurement.Input)
+	var port = ""
 	// allow URL input
 	if u, err := url.ParseRequestURI(host); err == nil {
-		host = u.Host
+		host = u.Hostname()
+		port = u.Port()
 	}
-	service := net.JoinHostPort(host, m.config.port())
-	udpAddr, err := net.ResolveUDPAddr("udp4", service)
+	var service string
+	if port == "" {
+		service = net.JoinHostPort(host, m.config.port())
+	} else {
+		service = net.JoinHostPort(host, port)
+	}
```

Comment:
I am going to test the diff with the following mechanical refactoring and merge with it if it looks good

```suggestion
	if port == "" {
		port = m.config.port()
	}
	service = net.JoinHostPort(host, port)
```